cogs/moderation.py

```diff
@@ -21,123 +53,54 @@ async def ping_library_developers(self, ctx, title, *, message):
-            db.modEntry.insert("Unmute (Server)", None, message, ctx.message.author.id, member.id,
-                               ctx.message.created_at)
-
         # If Library Developer invokes, unmute from channel and not server
         elif ((utils.get(ctx.guild.roles, name="Library Developer") in ctx.message.author.roles) and (
                 channel.category.name == "libraries" or channel.category.name == "frameworks")):
-            emb.add_field(name="Unmute (Channel)", value=f'{member.mention} ({member.id})', inline=False)
-            emb.add_field(name="Reason", value=f'{message}', inline=False)
-            emb.add_field(name="Moderator",
-                          value=f'{ctx.message.author}#{ctx.message.author.discriminator} ({ctx.message.author.id})',
-                          inline=False)
+            process_action(ctx, 'Unmute (Channel)', emb, member, message, None)
             await channel.set_permissions(member, overwrite=None)
 
-            # Add data to database
-            # (Action, Duration, Reason, Moderator, Target, Date)
-            db.modEntry.insert("Unmute (Channel)", None, message, ctx.message.author.id, member.id,
-                               ctx.message.created_at)
-
         await ctx.message.add_reaction('👍')
-        await log_channel.send(embed=emb)
+        await self.log_ch.send(embed=emb)
 
     @commands.command()
-    @commands.has_role("Library Developer" or "Moderator")
+    @commands.has_role("Moderator")
     async def warn(self, ctx, member: discord.Member, *, message):
-        channel = ctx.message.channel
-        log_channel = self.bot.get_channel(int(os.getenv("MODERATION_LOGS_CHANNEL")))  # moderation-logs channel
-
-        # Create default embed
-        emb = discord.Embed()
-        emb.set_author(name="Moderation",
-                       icon_url="https://cdn.discordapp.com/attachments/336577284322623499/683028692133216300/ac6e275e1f638f4e19af408d8440e1d1.png")
-        emb.set_footer(text=f'\t\t\t\t\t\t\tTimestamp: {ctx.message.created_at}')
-        emb.add_field(name="Warn", value=f'{member.mention} ({member.id})', inline=False)
-        emb.add_field(name="Reason", value=f'{message}', inline=False)
-        emb.add_field(name="Moderator",
-                      value=f'{ctx.message.author}#{ctx.message.author.discriminator} ({ctx.message.author.id})',
-                      inline=False)
-
-        # Add data to database
-        # (Action, Duration, Reason, Moderator, Target, Date)
-        db.modEntry.insert("Warn", None, message, ctx.message.author.id, member.id, ctx.message.created_at)
+        _, emb, channel = prepare_action(ctx, '')
+        process_action(ctx, 'Warn', emb, member, message, None)
 
         await ctx.message.add_reaction('👍')
-        await log_channel.send(embed=emb)
+        await self.log_ch.send(embed=emb)
 
     @commands.command()
-    @commands.has_role("Library Developer" or "Moderator")
+    @commands.has_role("Moderator")
```

Comment:
Library Developers can see infractions. This was done on purpose.

Reply:
e16c1b4